src/protocol.h

```diff
@@ -386,7 +386,7 @@ class CAddress : public CService
         int nVersion = s.GetVersion();
         if (s.GetType() & SER_DISK)
             READWRITE(nVersion);
-        if ((s.GetType() & SER_DISK) || !(s.GetType() & SER_GETHASH))
+        if ((s.GetType() & SER_DISK) || (nVersion >= CADDR_TIME_VERSION && !(s.GetType() & SER_GETHASH)))
```

Comment:
I guess that now we need to remove `CADDR_TIME_VERSION` from the "unused network protocol versions" section and put it back where it was.